src/contacts/tests/test_contacts.py

```diff
@@ -0,0 +1,333 @@
+        # mock return value of get_contacts to return EXAMPLE_CONTACT
+        self.mocked_db.return_value.get_contacts.return_value = [
+            EXAMPLE_CONTACT
+        ]
+        # create example contact request
+        example_contact = EXAMPLE_CONTACT.copy()
+        # change label of EXAMPLE_CONTACT while keeping other values
+        example_contact["label"] = "newlabel"
+        # send request to test client
+        response = self.test_app.post(
+            "/contacts/{}".format(EXAMPLE_USER),
+            headers=EXAMPLE_HEADERS,
+            data=json.dumps(example_contact),
+        )
+        # assert 409 response code
+        self.assertEqual(response.status_code, 409)
+        # assert we get correct error message
+        self.assertEqual(
+            response.json["msg"], "account already exists as a contact"
+        )
+
+    def test_create_contact_409_status_code_duplicate_contact_with_same_label(self,):
+        """test adding a duplicate contact with same label, different account/routing num"""
+        # mock return value of get_contacts to return EXAMPLE_CONTACT
+        self.mocked_db.return_value.get_contacts.return_value = [
+            EXAMPLE_CONTACT
+        ]
+        # create example contact request
+        example_contact = EXAMPLE_CONTACT.copy()
+        # change account_num and routing_num
+        example_contact["account_num"] = "1231231231"
+        example_contact["routing_num"] = "123123123"
+        # send request to test client
+        response = self.test_app.post(
+            "/contacts/{}".format(EXAMPLE_USER),
+            headers=EXAMPLE_HEADERS,
+            data=json.dumps(example_contact),
+        )
+        # assert 409 response code
+        self.assertEqual(response.status_code, 409)
+        # assert we get correct error message
+        self.assertEqual(
+            response.json["msg"], "contact already exists with that label"
+        )
+
+    def test_create_contact_400_status_code_invalid_account_number_less_than_ten_digits(self,):
+        """test adding a contact with invalid account number"""
+        # create example contact request
+        example_contact = EXAMPLE_CONTACT.copy()
+        # change account_num to 9 digits
+        example_contact["account_num"] = "123123123"
+        # send request to test client
+        response = self.test_app.post(
+            "/contacts/{}".format(EXAMPLE_USER),
+            headers=EXAMPLE_HEADERS,
+            data=json.dumps(example_contact),
+        )
+        # assert 400 response code
+        self.assertEqual(response.status_code, 400)
+        # assert we get correct error message
+        self.assertEqual(response.json["msg"], "invalid account number")
+
+    def test_create_contact_400_status_code_invalid_routing_number_more_than_nine_digits(self,):
+        """test adding a contact with invalid routing number"""
+        # create example contact request
+        example_contact = EXAMPLE_CONTACT.copy()
```

Comment:
same as previous comments, extending out comment on var names and what a helper method could look like
```
invalid_routing_num = ""1231231231""
invalid_contact = newContact(routing_num=invalid_routing_num)
````

Reply:
Done